File: Lecture6-Python/pset6/sentimental-readability/readability.py
# TODO
import cs50
import math
import sys

def main():
    text = cs50.get_string("Text: ")

    record = get_all_count(text)
    
    grade = get_grade(record)
    
    if grade < 1:
        print("Before Grade 1")
        sys.exit(0)
    if grade >= 16:
        print("Grade 16+")
        sys.exit(0)
    
    print(f"Grade {grade}")
    sys.exit(0)

def get_all_count(text):
    record = {}
    
    words = 1
    sentences = 0
    letters = 0
    for word in text:
        if word.isalpha():
            letters += 1
        if word  == " ":
            words += 1
        if word == '!' or word == '?' or word == '.':
            sentences += 1
    
    record["words"] = words
    record["sentences"] = sentences
    record["letters"] = letters
    
    return record

def get_grade(record):
    L = float(record["letters"]/record["words"]) * 100 
    S = float(record["sentences"]/record["words"]) * 100
    grade = round(0.0588 * L - 0.296 * S - 15.8)
    return grade

main()

File: Lecture2-Arrays/pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int letters = 0;
int words = 1;
int sentences = 0;

void init_everything(string text);
double get_grade(int letters, int sentences, int words);

int main(void) 
{
    // get user input
    string text = get_string("Text: ");

    // calculate the number of letters, sentences and words
    init_everything(text);

    // calculate grade
    double grade = get_grade(letters, sentences, words);

    // print the output
    if (grade < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    printf("Grade %1.0f\n", round(grade));
}

void init_everything(string text)
{
    for (int i = 0, length=strlen(text); i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
        if (isblank(text[i]))
        {
            words += 1;
        }
        if (ispunct(text[i]))
        {
            if (text[i] == '.' || text[i] == '?' || text[i] == '!')
            {
                sentences += 1;
            }
        }
    }
}

double get_grade(int letters, int sentences, int words)
{
    float L = (float)letters/words * 100;
    float S = (float)sentences/words * 100;
    double grade = ((0.0588 * L) - (0.296 * S)) - (float)15.8;
    printf("%f, %f, %f\n", L, S, grade);
    return grade;
}